lib/jwt.rb

```diff
@@ -37,10 +45,22 @@ def sign_rsa(algorithm, msg, private_key)
     private_key.sign(OpenSSL::Digest.new(algorithm.sub('RS', 'sha')), msg)
   end
 
+  def sign_ecdsa(algorithm, msg, private_key)
+    raise IncorrectAlgorithm unless NAMED_CURVES[algorithm] == private_key.group.curve_name
```

Comment:
Now that I look back at this. It would probably be better to raise this with a more helpful error.